src/schema/field_type.rs

```diff
@@ -58,13 +65,14 @@ pub enum Type {
     Json = b'j',
 }
 
-const ALL_TYPES: [Type; 9] = [
+const ALL_TYPES: [Type; 10] = [
     Type::Str,
     Type::U64,
     Type::I64,
     Type::F64,
     Type::Bool,
     Type::Date,
```

Comment:
The old Date type was actually a `DateTime` (yes the name sucks).

Do we plan to remove it?

Reply:
So all these two types `DateTime` and `PreciseDateTime` exist because my plan is to keep Type::Date working with a deprecated label. If we can drop the old DateTime type and accept the breaking change let me know your opinion.